notebook-src/02-book-balancing-intro.rst

```diff
@@ -0,0 +1,298 @@
+   2.222222    0.795220          -0.606321            0.795220
+   2.777778    0.355842          -0.934546            0.355842
+   3.333333   -0.190568          -0.981674           -0.190568
+   3.888889   -0.679658          -0.733529           -0.679658
+   4.444444   -0.964317          -0.264750           -0.964317
+   5.000000   -0.958924           0.283662           -0.958924
+
+The result of the last simulation is always stored on the system for later
+use::
+
+   >>> sys.last_trajectories
+             book_angle      mass_center_y  mass_center_height
+   time
+   0.000000    0.000000           1.000000            0.000000
+   0.555556    0.527415           0.849608            0.527415
+   1.111111    0.896192           0.443666            0.896192
+   1.666667    0.995408          -0.095724            0.995408
+   2.222222    0.795220          -0.606321            0.795220
+   2.777778    0.355842          -0.934546            0.355842
+   3.333333   -0.190568          -0.981674           -0.190568
+   3.888889   -0.679658          -0.733529           -0.679658
+   4.444444   -0.964317          -0.264750           -0.964317
+   5.000000   -0.958924           0.283662           -0.958924
+
+We can now plot these variables, one at a time::
+
+   >>> %matplotlib inline
+   >>> trajectories['book_angle'].plot()
+
+altogether::
+
+   >>> trajectories.plot()
+
+or in subplots::
+
+   >>> trajectories.plot(subplots=True)
+
+Maybe you want to use degrees instead, just make a new column::
+
+   >>> trajectories['book_angle_deg'] = np.rad2deg(trajectories['book_angle'])
+   >>> trajectories['book_angle_deg'].plot()
+
+Exercise
+--------
+
+Simulate the system with different initial conditions and parameter values.
+
+- Does the simulation always work, if not what doesn't work? *Hint: try a tall
+  stack of books, can you find a stack height that is significant?*
+- Are there any mathematical functions that could be used describe the change
+  in the book angle?
+
+Animate The Motion
+==================
+
+Plotting the coordinates and measurements as a function with respect to time is
+a very useful way to visualize a system's motion, but it is often quite helpful
+to animate a pictorial diagram of the system for easier visualization of the
+motion. matplotlib has
+
+::
+
+   >>> import matplotlib.pyplot as plt
+   >>> from matplotlib.patches import Circle, Rectangle
+   >>> import matplotlib.animation as animation
+   >>> def setup_figure():
+           fig, ax = plt.subplots(1, 1)
+           circ = Circle((0, radius), radius)
+           rect = Rectangle((mass_center_x, mass_center_y), width, height)
+           ax.add_patch(circ)
+           ax.add_patch(rect)
+   >>> def animate(i):
+           rect.set_xy(mass_center_x, mass_center_y)
+           rect.set_rotation(book_angle)
+   >>> sys.animation_setup_func = setup_figure
+   >>> sys.animation_update_func = animate
+   >>> sys.animate()
```

Comment:
I think this might work the same as the measurement functions. Basically you can create measurements for any important points for plotting and then use them directly in these two functions. Making it easy to add an animation to a system.